alt-fuel-finder/app/models/station.rb

```diff
@@ -1,6 +1,34 @@
 class Station
+  attr_reader :service
+  Station = Struct.new("Station",
+                      :id,
+                      :fuel_type_code,
+                      :station_phone,
+                      :city,
+                      :intersection_directions,
+                      :street_address,
+                      :fuel_type_code,
+                      :access_days_time,
+                      :distance
+                      )
 
-  def initialize(params)
+  def self.service
+    @service ||= NrelService.new
   end
 
+  def self.nearest_stations_by_zip(zip_code, station_count = 10)
+    service.nearest_stations_by_zip(zip_code).map do |station|
+      Station.new(
+                  station[:id],
+                  station[:fuel_type_code],
+                  station[:station_phone],
+                  station[:city],
+                  station[:intersection_directions],
+                  station[:street_address],
+                  station[:fuel_type_code],
+                  station[:access_days_time],
+                  station[:distance]
+                  )
+    end
+  end
 end
```

Comment:
My first commit on this refactor [here](https://github.com/josh-works/module_3_diagnostics/pull/1/commits/1083476320671640176c0bf9a8cba01b5d290c74#diff-7b892454fb26ac38b70aa25f877d3735) used OpenStruct to create the station objects.

I know that's not ideal, so I am now using Struct to accomplish the same goal.